kobotoolbox/kpi
# Show 'SUMMARY' & 'DATA' tabs (with filtered data) to users with 'partial_view_submissions' permission.

It covers `FE` & `BE` issues. 
Fixes #2385 

## Test procedure 
I followed these steps to ensure everything was working as expected.

- Go to `kpi` admin platform. 
- Create two new users `User A` and `User B`
- Log into `User A`'s account
- Go to `Accounts settings` and check `Require authentication to see forms and submit data`
- Create a new form 
- Assign `Partial View Submissions (for `User B`), Add Submissions` to `User B`. 
- Open the form in Enketo 
- Submit data once with `User A`'s credentials.
- Log into `User B`'s account (in `kpi`) with another browser.
- Open the form in Enteko. Be sure to log out if you are already logged in.
- Submit data once with `User B`'s credentials.

`User A` should see 2 submissions in:
- Summary
- Reports 
- Table View
- Exports

_Maps & Gallery depend on the type of questions you added in the form_

`User B` should see only 1 submission in:
- Summary
- Reports 
- Table View (Number of submissions is wrong: Read below `know issues`)
- Exports

_Maps depends on the type of questions you added in the form_
**Gallery and legacy exports can't work** because `kc` doesn't know anything about `partial_view_submission` permission. iframe shows `Not shared`. @jnm suggested to hide those options. 

N.B.: I've also follow the same procedure with the registration form and two other new users.

## Known issues

1. ~~Total of submissions is wrong in the header for users with `partial_view_submission` permission:~~
    ~~`FE` gets value from the asset's `deployment__submission_count` property which comes from `XForm`.num_of_submissions property.~~
    - ~~https://github.com/kobotoolbox/kpi/blob/2385-partial-submission-permissions-fixes/kpi/serializers/v2/asset.py#L256-L259~~
    - ~~https://github.com/kobotoolbox/kpi/blob/2385-partial-submission-permissions-fixes/kpi/deployment_backends/base_backend.py#L110-L112~~
    - ~~https://github.com/kobotoolbox/kpi/blob/2385-partial-submission-permissions-fixes/kpi/deployment_backends/kobocat_backend.py#L423-L430~~
    - ~~https://github.com/kobotoolbox/kpi/blob/2385-partial-submission-permissions-fixes/kpi/deployment_backends/kc_access/utils.py#L46-L53~~
    
    ~~It can be easily fixed in `kpi` by querying mongo with correct filters but it can be a performance issue if there is lots of instances.~~ 

     Fixed in this PR.
     Users can still read `kc` value from `kc`'s API.

2. Number of submissions is wrong in Table View. 
     It's fixed within PR #2331
